common/concurrency/ConcurrentQueue.h

```diff
@@ -94,6 +94,13 @@ class ConcurrentUnBoundedQueue : public AbstractConcurrentBoundedQueue<Elem, moo
 template <class Elem>
 using BlockingBoundedQueue = AbstractConcurrentBoundedQueue<Elem, moodycamel::BlockingConcurrentQueue<Elem>>;
 
+template <class Elem>
+class BlockingUnBoundedQueue : public AbstractConcurrentBoundedQueue<Elem, moodycamel::BlockingConcurrentQueue<Elem>> {
+public:
+    BlockingUnBoundedQueue()
+        : AbstractConcurrentBoundedQueue<Elem, moodycamel::BlockingConcurrentQueue<Elem>>(INT_MAX){};
```

Comment:
OOC: how many messages do we get per second? 
How long would it take an engineer to reach INT_MAX?
(previously it was used for number of files and was effectively non-reachable, but with number of IDE interactions, I'd like to confirm)

Reply:
This isn't the perfect methodology, but I looked in Splunk to figure this out.

The most active Sorbet user over the past hour sent 1,510 messages to Sorbet. Over the past 15 minutes, it was 457 messages. (This is much more active than other users.)

Let's assume that all of those are edits (didChange and such), and that VS Code will send document symbol, code action, and completion requests along with each edit when those features stabilize. Let's also assume that a high-activity session is double the number of messages -- ~1 edit message per second -- which translates into 4 messages per second with the other requests.

If MAX_INT is the maximum positive 32-bit signed integer (2,147,483,647), then we could sustain a high-activity Sorbet session for ~17 years.